app/src/main/java/com/malmstein/yahnac/stories/ArticleActivity.java

```diff
@@ -140,11 +146,13 @@ private void uncheckBookmarkMenuItem(MenuItem bookmarks) {
 
     private void removeBookmark(DataPersister persister, Story story) {
         persister.removeBookmark(story);
+        uncheckBookmarkMenuItem(getBookmarkMenuItem());
```

Comment:
You don't need this. Just call `invalidateOptionsMenu()`: the method in turn will call `onCreateOptionsMenu()` again and `checkBookmarkMenuItem()/uncheckBookmarkMenuItem` inside it.